decorators/celery_.py

```diff
@@ -10,12 +11,24 @@
 from htk.utils.notifications import slack_notify
 from htk.utils.timer import HtkTimer
 
+# Zippy Imports
+from zippy.celeryconfig import app as celery_app
+
 
 class safe_timed_task(object):
     def __init__(self, task_name, notify=False):
         self.task_name = task_name
         self.notify = notify
 
+        current_time = datetime.now()
+        formatted_time = current_time.strftime('%Y-%m-%d %H:%M:%S')
+        self.time_created = formatted_time
+
+        with celery_app.connection() as connection:
+            queue = connection.default_channel.queue_declare(queue='zippy_prod_celery', passive=True)
+            queue_position = queue.message_count + 1
+        self.queue_position = queue_position
```

Comment:
Instead of this addition, you might want to create a separate class called `safe_timed_tasks_with_queue_position` that extends from `safe_timed_task` class and let it accept `celery_app` in its `__init__` function as a parameter. That way dependency can be provided where it is being used.